### PR TITLE
[sdk/python] Add ready attribute to await Helm charts

### PR DESCRIPTION
**Pull request by lblackstone**
### Related issues (optional)

Part of #861

### CODE REVIEW

File: tests/sdk/python/helm-local/step1/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-pulumi>=2.0.0,<3.0.0
+pulumi>=3.0.0,<4.0.0
```
Comment (viveklak) [inline tests/sdk/python/helm-local/step1/requirements.txt:1]: Does this need to be >= 3.9.0 as well?

File: tests/sdk/python/python_test.go [hunk 2]
```diff
@@ -360,6 +361,34 @@ func TestHelmLocal(t *testing.T) {
 	}
 	options := baseOptions.With(integration.ProgramTestOptions{
 		Dir: filepath.Join(cwd, "helm-local", "step1"),
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
```
Comment (viveklak) [inline tests/sdk/python/python_test.go:364]: This is neat! DId you verify this fails before the fix?

**Review by viveklak — APPROVED** [review]

File: tests/sdk/python/python_test.go (hunk 2 above)
Comment (lblackstone) [reply tests/sdk/python/python_test.go:364]: Yep! Without the fix, both ConfigMap resources will create before the Deployment and Service.

File: tests/sdk/python/helm-local/step1/requirements.txt (hunk 1 above)
Comment (lblackstone) [reply tests/sdk/python/helm-local/step1/requirements.txt:1]: Hmm, must be getting overridden somewhere, but I'll update.

**Pull request merged**